providers/oidc.go

```diff
@@ -285,8 +287,14 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
 	groups := []string{}
 
-	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
-	if rawGroups != nil && ok {
+	switch rawGroups := rawClaims[p.GroupsClaim].(type) {
+	case []string:
+		groups = append(groups, rawGroups...)
+	case string:
+		groups = append(groups, rawGroups)
+	case int:
+		groups = append(groups, strconv.Itoa(rawGroups))
+	case []interface{}:
```

Comment:
All this should be handled by `formatGroup` since it has the logic to handle string vs non-string (including the int case).

So I think this just needs to handle single vs multi in how it calls `formatGroup`

Reply:
Looking at `formatGroup` it seems to marshal into a JSON string, so in theory a JSON hashmap string could be returned. Will that work out in the downstream usage of the group values? I removed the `int` check but it looks like all other cases would be handled just as well?

Reply:
I think we just want to handle `[]interface{}` vs `interface{}` in the switch and let `formatGroup` do all the work on the underlying data in the list or singleton (string, number, complex object, whatever).

Reply:
@NickMeves hm I'm kind of lost, since `formatGroup` will return a JSON string should I then go on and parse this?

Reply:
`formatGroup` already handles if it is a `string` or not and returns the plain string with no JSON unmarshalling if it is a string: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/providers/oidc.go#L305

So you don't need to differentiate `string` vs `interface` in your switch, `formatGroup` does that. You just need to handle singular vs list, since the old code assumed everything was a list.

And for non string cases, it will handle number & complex based cases.  You should add back the numeric test you previously had though to make sure the number case does JSON unmarshal correctly to a string through.